mlflow/projects/__init__.py

```diff
@@ -781,7 +781,8 @@ def _get_databricks_env_vars(tracking_uri):
         return {}
 
     db_profile = get_db_profile_from_uri(tracking_uri)
-    config = databricks_utils.get_databricks_host_creds(db_profile)
+    db_path_info = get_db_path_info_from_uri(tracking_uri)
```

Comment:
I changed all the locations where we get the host creds by default, but I'm not sure that we actually need this everywhere. We definitely need it in the tracking and registry utils to fetch artifacts and to do all other registry operations, but I honestly don't have enough context on projects to know if it's right to put it here

Reply:
hm yeah. i think it'd make sense to support it here. how about if we made `get_databricks_host_creds ` take in the URI instead, since all callsites essentially have and use the URI anyway? In `mlflow/projects/databricks.py` there are a few places where we store `databricks_profile` and use it to call `get_databricks_host_creds` but it doesn't seem like we use it for anything else, so if we stored the URI itself instead of the parsed profile name I don't think there would be a problem.

cc @smurching - in projects, `DatabricksSubmittedRun` and `DatabricksJobRunner` use `databrick_profile` to get the hostcreds. if we change the `DatabricksJobRunner.databricks_profile` field from a string to the tracking URI itself (and possibly rename it), do you think there would be a problem? those objects seem to be used just within `mlflow/projects/databricks.py` (but i don't know if people have plug-ins or anything)

Reply:
Yeah that would make sense @sueann ! Will wait until @smurching weighs in before I do the refactor

Reply:
They're internal classes, seems fine to refactor :)